cylc/flow/xtrigger_mgr.py

```diff
@@ -159,6 +164,244 @@ class TemplateVariables(Enum):
 RE_STR_TMPL = re.compile(r'(?<!%)%\(([\w]+)\)s')
 
 
+class XtriggerCollator:
+    """Collate and validate parsed xtriggers.
+
+    Separate from XtriggerManager to simplify validation and testing.
+
+    """
+
+    def __init__(self):
+        # Map xtrig label to function context.
+        self.functx_map: 'Dict[str, SubFuncContext]' = {}
+        # Clock labels, to avoid repeated string comparisons
+        self.wall_clock_labels: Set[str] = set()
+        # Workflow-wide default, used when not specified in xtrigger kwargs.
+        self.sequential_xtriggers_default = False
+        # Labels whose xtriggers are sequentially checked.
+        self.sequential_xtrigger_labels: Set[str] = set()
+
+    def update(self, xtriggers: 'XtriggerCollator'):
+        self.functx_map.update(xtriggers.functx_map)
+        self.wall_clock_labels.update(xtriggers.wall_clock_labels)
+        self.sequential_xtrigger_labels.update(
+            xtriggers.sequential_xtrigger_labels)
+
+    def add_trig(self, label: str, fctx: 'SubFuncContext', fdir: str) -> None:
+        """Add a new xtrigger function.
+
+        Args:
+            label: xtrigger label
+            fctx: function context
+            fdir: module directory
+
+        """
+        if label in self.functx_map:
+            # we've already seen this one
+            return
+
+        if (
+            not label.startswith('_cylc_retry_') and not
+            label.startswith('_cylc_submit_retry_')
+        ):
```

Comment:
```suggestion
        if fctx.func_name != _wall_clock.__name__:
```

Reply:
Actually, I had to revert your suggestion - it doesn't cover all the bases:

```
FAILED tests/integration/test_simulation.py::test_settings_reload - cylc.flow.exceptions.XtriggerConfigError: [@_cylc_retry_1066/one] wall_clock(trigger_time=1718159104.6357827)
got an unexpected keyword argument 'trigger_time'
```